src/modules/launcher/Wox.Infrastructure/Image/ImageCache.cs

```diff
@@ -22,7 +23,26 @@ public class ImageCache
                 var i = _data[path];
                 return i;
             }
-            set { _data[path] = value; }
+            set 
+            {
+                _data[path] = value;
+
+                // To prevent the dictionary from drastically increasing in size by caching images, the dictionary size is not allowed to grow more than the permissibleFactor * maxCached size
+                // This is done so that we don't constantly perform this resizing operation and also maintain the image cache size at the same time
+                if (_data.Count > permissibleFactor * MaxCached)
+                {
+                    Cleanup();
+                    foreach (var key in _data.Keys)
```

Comment:
[nit] Maybe add a comment to explain this part so that anyone reading this part doesn't need to go through Cleanup's definition to understand why this loop removes any entries that are not in the top used `maxCached` images.